### PR TITLE
Changed the deleteTicket(String) method on the org.jasig.cas.ticket.r…

### LINKED ISSUE #1969: STs that are attached to a TGT are not counted in total ticket cleanup
...while using the default ticket registry. This results in inconsistent ticket count reporting while cleaning.
Cleanup for 5 separate logins (5 TGTs + 5 STs), cleaner reports ~7-9 tickets cleaned.

Re-attaching the log file for reference.
[cas.zip](https://github.com/apereo/cas/files/437531/cas.zip)

CAS 4.2.5-SNAPSHOT
Tomcat 7, Java 8.
https://groups.google.com/a/apereo.org/forum/#!topic/cas-user/ZMerMRq_EWg

### PR DESCRIPTION
**Pull request by wcrowell**
Closes #1969 

- Changed org.apereo.cas.ticket.registry.TicketRegistry's deleteTicket method to return the total count of tickets deleted which includes any child tickets.
- Unit tests have been updated to account for this change.

### CODE REVIEW

File: cas-server-core-tickets/src/test/java/org/apereo/cas/ticket/registry/DistributedTicketRegistryTests.java [hunk 1]
```diff
@@ -102,6 +106,34 @@ public void verifyTicketDoesntExist() {
...
+        assertNotNull(this.ticketRegistry.getTicket("ST1", ServiceTicket.class));
+
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+
+        assertSame(3, this.ticketRegistry.deleteTicket(tgt.getId()));
+
+        assertNull(this.ticketRegistry.getTicket(TGT_NAME, TicketGrantingTicket.class));
+        assertNull(this.ticketRegistry.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.ticketRegistry.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-core-tickets/src/test/java/org/apereo/cas/ticket/registry/DistributedTicketRegistryTests.java]: This should `ProxyGrantingTicket.class`

File: cas-server-support-ehcache-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/EhCacheTicketRegistryTests.java [hunk 2]
```diff
@@ -241,14 +242,41 @@ public void verifyDeleteTicketWithChildren() {
...
+        assertNotNull(this.ticketRegistry.getTicket("ST1", ServiceTicket.class));
+
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+
+        assertSame(3, this.ticketRegistry.deleteTicket(tgt.getId()));
+
+        assertNull(this.ticketRegistry.getTicket("TGT", TicketGrantingTicket.class));
+        assertNull(this.ticketRegistry.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.ticketRegistry.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-support-ehcache-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/EhCacheTicketRegistryTests.java]: See my comment above

File: cas-server-support-hazelcast-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/HazelcastTicketRegistryTests.java [hunk 3]
```diff
@@ -117,14 +118,43 @@ public void verifyDeleteTicketWithChildren() throws Exception {
...
+
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+
+        this.hzTicketRegistry1.updateTicket(tgt);
+        assertSame(3, this.hzTicketRegistry1.deleteTicket(tgt.getId()));
+
+        assertNull(this.hzTicketRegistry1.getTicket("TGT", TicketGrantingTicket.class));
+        assertNull(this.hzTicketRegistry1.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.hzTicketRegistry1.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-support-hazelcast-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/HazelcastTicketRegistryTests.java]: Same here

File: cas-server-support-ignite-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/IgniteTicketRegistryTests.java [hunk 4]
```diff
@@ -241,14 +242,42 @@ public void verifyDeleteTicketWithChildren() {
...
+
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+        
+        this.ticketRegistry.updateTicket(tgt);
+        assertSame(3, this.ticketRegistry.deleteTicket(tgt.getId()));
+        
+        assertNull(this.ticketRegistry.getTicket("TGT", TicketGrantingTicket.class));
+        assertNull(this.ticketRegistry.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.ticketRegistry.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-support-ignite-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/IgniteTicketRegistryTests.java]: And here

File: cas-server-support-infinispan-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/InfinispanTicketRegistryTests.java [hunk 5]
```diff
@@ -76,4 +80,31 @@ private static Ticket getTicket() {
...
+        assertNotNull(this.infinispanTicketRegistry.getTicket(TGT_NAME, TicketGrantingTicket.class));
+        assertNotNull(this.infinispanTicketRegistry.getTicket("ST1", ServiceTicket.class));
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+
+        assertTrue("TGT and children were deleted", this.infinispanTicketRegistry.deleteTicket(tgt.getId()) == 3);
+
+        assertNull(this.infinispanTicketRegistry.getTicket(TGT_NAME, TicketGrantingTicket.class));
+        assertNull(this.infinispanTicketRegistry.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.infinispanTicketRegistry.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-support-infinispan-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/InfinispanTicketRegistryTests.java]: And here

File: cas-server-support-memcached-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/MemCacheTicketRegistryTests.java [hunk 6]
```diff
@@ -136,4 +134,31 @@ public void verifyDeleteTicketWithChildren() throws Exception {
...
+        assertNotNull(this.registry.getTicket("ST1", ServiceTicket.class));
+
+        final TicketGrantingTicket pgt = st1.grantProxyGrantingTicket("PGT-1", a, new NeverExpiresExpirationPolicy());
+        assertEquals(a, pgt.getAuthentication());
+        
+        assertSame(3, this.registry.deleteTicket(tgt.getId()));
+        
+        assertNull(this.registry.getTicket("TGT", TicketGrantingTicket.class));
+        assertNull(this.registry.getTicket("ST1", ServiceTicket.class));
+        assertNull(this.registry.getTicket("PGT-1", ServiceTicket.class));
```
Comment (mmoayyed) [inline cas-server-support-memcached-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/MemCacheTicketRegistryTests.java]: And finally here.

**Comment by wcrowell** [comment]
Fixed.

**Pull request merged**